### PR TITLE
refactor(core): Bifurcate handling of webhooks vs. triggers and pollers (no-changelog)

### PR DESCRIPTION
**Pull request by ivov**
As our understanding of multi-main setup [continues to evolve](https://excalidraw.com/#json=ExMQqdHx4lW3BSazPhxPb,pm8GjGqW-iv2gF-xVoDlww), this PR again refactors all flows to match what we currently think is the best approach.

- **Changes to init flows:** On init, only the leader should add triggers and pollers. On init, neither leader nor follower should add webhooks, in line with skipping webhook deregistration on shutdown. No changes to `leadershipChange` init.
- **Changes to update flows:** On update, the main that received the workflow update request, whether leader or follower, should add or remove webhooks, and send a message for the leader to add or remove triggers and pollers. The prior approach of having only the leader remove webhooks fails to deregister webhooks in cases where e.g. we replace an active webhook with an active trigger without deactivation. Note that in this setup a main should be able to message itself, as opposed to before where only followers could message the leader.
- **Changes to event handlers:** On `leader-stepdown` (former leader realizes they lost leadership), the former leader should remove all triggers and pollers for all workflows and stop pruning. On `leader-takeover` (follower became leader), the new leader should add all triggers and pollers for all workflows and start pruning. These events used to be `leadershipVacant` and `leadershipChange` with slightly different effects.

Notes:
- Switched some logs to `warn` to make them easier to scan while debugging. Will revert these before merging.
- Passing `versionId` to every message when adding webhooks, triggers and pollers increases complexity in the already complex `add` and `remove` operations. `versionId` is only needed in case we need to deactivate a workflow that failed to activate in the leader in reaction to a message. Is it safe to deactivate on failure without referencing the `versionId`?
- Tore down all obsolete tests from the prior approach, as they are no longer relevant to this setup. Coming up with a testing approach that accounts for this new setup is more time-consuming than the time available skipping this for now. Did some basic testing - will test all variants listed in the [battery of manual tests](https://github.com/n8n-io/n8n/pull/8446) after review, as discussed. Let me know otherwise.

### CODE REVIEW

File: packages/cli/src/ActiveWorkflowRunner.ts [hunk 1]
```diff
@@ -388,7 +389,7 @@ export class ActiveWorkflowRunner {
 				responsePromise?: IDeferredPromise<IExecuteResponsePromiseData>,
 				donePromise?: IDeferredPromise<IRun | undefined>,
 			): void => {
-				this.logger.debug(`Received trigger for workflow "${workflow.name}"`);
+				this.logger.warn(`Received trigger for workflow "${workflow.name}"`);
```
Comment (krynble) [inline packages/cli/src/ActiveWorkflowRunner.ts:392]: This change will make the logs quite verbose IMO

File: packages/cli/src/ActiveWorkflowRunner.ts [hunk 2]
```diff
@@ -768,6 +791,21 @@ export class ActiveWorkflowRunner {
 		}
 	}
 
+	/**
+	 * Stop running active triggers and pollers for a workflow.
+	 */
+	async removeWorkflowTriggersAndPollers(workflowId: string) {
```
Comment (krynble) [inline packages/cli/src/ActiveWorkflowRunner.ts:797]: Maybe we can deduplicate the `remove` function above to use this function as well wdyt?

File: packages/cli/src/commands/start.ts
Comment (krynble) [inline packages/cli/src/commands/start.ts:1]: Looks so much nicer now :D

File: packages/cli/src/ActiveWorkflowRunner.ts (hunk 1 above)
Comment (ivov) [reply packages/cli/src/ActiveWorkflowRunner.ts:392]: Did this on purpose - this highlighting makes finding relevant logs while testing much easier. They're only temporary, see above:

> Switched some logs to warn to make them easier to scan while debugging. Will revert these before merging.

**Review by krynble — APPROVED** [review]

**Pull request merged**